lib/smart_answer_flows/marriage-abroad.rb

```diff
@@ -126,6 +126,13 @@ def define
           elsif calculator.partner_is_opposite_sex?
             if calculator.ceremony_country == 'hong-kong'
               outcome :outcome_opposite_sex_marriage_in_hong_kong
+            elsif calculator.ceremony_country == 'south-africa' &&
+                (
+                  calculator.resident_of_uk? ||
+                  calculator.resident_of_ceremony_country? ||
+                  calculator.resident_of_third_country?
+                )
```

Comment:
Is this check needed? A user must always be either a resident of the UK, a resident of South Africa or a resident of another country, so this check will always be truthy.